datatune/core/filter.py

```diff
@@ -8,18 +8,22 @@
 import ast
 
 
-def input_as_string(serialized_input_column: str, df: pd.DataFrame) -> pd.DataFrame:
+def input_as_string(serialized_input_column: str, df: pd.DataFrame, input_fields:Optional[List[str]] = None) -> pd.DataFrame:
     """
     Converts each row in the DataFrame to a string representation and stores it in a new column.
 
     Args:
         serialized_input_column (str): Name of the column to store the serialized row data.
         df (pd.DataFrame): Input DataFrame to process.
+        input_fields (Optional[List], optional): List of input fields to include. Defaults to None.
 
     Returns:
         pd.DataFrame: DataFrame with the added serialized input column.
     """
-    df[serialized_input_column] = [str(row.to_dict()) for _, row in df.iterrows()]
+    if input_fields:
+        df[serialized_input_column] = [str({k:v for k,v in row.to_dict().items() if k in input_fields}) for _, row in df.iterrows()]
```

Comment:
make a new df view like `df = df with only input_fields columns` then `df[serialized_input_column] = [str(row.to_dict()) for _, row in df.iterrows()]`